--- qt-everywhere-opensource-src-4.8.6/src/plugins/accessible/widgets/itemviews.cpp.omv~	2014-06-07 12:26:59.253966245 +0200
+++ qt-everywhere-opensource-src-4.8.6/src/plugins/accessible/widgets/itemviews.cpp	2014-06-07 12:28:15.563012251 +0200
@@ -393,7 +393,7 @@ bool QAccessibleTable2::unselectColumn(i
     QModelIndex index = view()->model()->index(0, column, view()->rootIndex());
     if (!index.isValid() || view()->selectionMode() & QAbstractItemView::NoSelection)
         return false;
-    view()->selectionModel()->select(index, QItemSelectionModel::Columns & QItemSelectionModel::Deselect);
+    view()->selectionModel()->select(index, (QItemSelectionModel::SelectionFlags)(QItemSelectionModel::Columns & QItemSelectionModel::Deselect));
     return true;
 }
 
--- qt-everywhere-opensource-src-4.8.6/src/gui/kernel/qx11embed_x11.cpp.omv~	2014-06-08 08:34:49.547537784 +0000
+++ qt-everywhere-opensource-src-4.8.6/src/gui/kernel/qx11embed_x11.cpp	2014-06-08 08:35:20.753147660 +0000
@@ -275,7 +275,7 @@ public:
     QTLWExtra* topData() { return d_func()->topData(); }
 };
 
-static unsigned int XEMBED_VERSION = 0;
+static constexpr long XEMBED_VERSION = 0;
 
 enum QX11EmbedMessageType {
     XEMBED_EMBEDDED_NOTIFY = 0,
